src/System.Private.ServiceModel/src/System/IdentityModel/Selectors/KerberosSecurityTokenProvider.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IdentityModel.Tokens;
+using System.Net;
+using System.Security.Authentication.ExtendedProtection;
+using System.Security.Principal;
+using System.ServiceModel;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IdentityModel.Selectors
+{
+    public class KerberosSecurityTokenProvider : SecurityTokenProvider
+    {
+        private readonly string _servicePrincipalName;
+        private readonly TokenImpersonationLevel _tokenImpersonationLevel;
+        private readonly NetworkCredential _networkCredential;
+
+        public KerberosSecurityTokenProvider(string servicePrincipalName)
+            : this(servicePrincipalName, TokenImpersonationLevel.Identification)
+        {
+        }
+
+        public KerberosSecurityTokenProvider(string servicePrincipalName, TokenImpersonationLevel tokenImpersonationLevel)
+            : this(servicePrincipalName, tokenImpersonationLevel, null)
+        {
+        }
+
+        public KerberosSecurityTokenProvider(string servicePrincipalName, TokenImpersonationLevel tokenImpersonationLevel, NetworkCredential networkCredential)
+        {
+            if (servicePrincipalName == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("servicePrincipalName");
+            if (tokenImpersonationLevel != TokenImpersonationLevel.Identification && tokenImpersonationLevel != TokenImpersonationLevel.Impersonation)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException("tokenImpersonationLevel",
+                    "blahblahblah"/*SR.Format(SR.ImpersonationLevelNotSupported, tokenImpersonationLevel)*/));
```

Comment:
Yep, that was put in there as the resources didn't exist! I forgot to fix it :smile: At least it was obvious that it was a missed item.